socorro/lib/search_common.py

```diff
@@ -182,16 +182,17 @@ def get_parameters(self, **kwargs):
                                 break
 
                     # ensure the right data type
-                    try:
-                        value = convert_to_type(value, param.data_type)
-                    except ValueError as exc:
-                        raise BadArgumentError(
-                            param.name,
-                            msg=(
-                                "Bad value for parameter %s: not a valid %s"
-                                % (param.name, param.data_type)
-                            ),
-                        ) from exc
+                    if operator != "__null__":
```

Comment:
In the case of NumberField and using `__null__`, the value is `__null__` and that's not a valid int. That's the only exception and it'll probably be an exception for other things, too, so I added a test here.